e2etests-cli/main_test.go

```diff
@@ -544,6 +550,169 @@ i used to read
 				assert.Equal(t, "x -> y\n", string(gotBar))
 			},
 		},
+		{
+			name: "watch-regular",
+			run: func(t *testing.T, ctx context.Context, dir string, env *xos.Env) {
+				writeFile(t, dir, "index.d2", `
+a -> b
+b.link: layers.cream
+
+layers: {
+    cream: {
+        c -> b
+    }
+}`)
+				stderr := &bytes.Buffer{}
+				tms := testMain(dir, env, "--watch", "--browser=0", "index.d2")
+				tms.Stderr = stderr
+
+				tms.Start(t, ctx)
+				defer func() {
+					// Manually close, since watcher is daemon
+					err := tms.Signal(ctx, os.Interrupt)
+					assert.Success(t, err)
+				}()
+
+				// Wait for watch server to spin up and listen
+				urlRE := regexp.MustCompile(`127.0.0.1:([0-9]+)`)
+				watchURL := waitLogs(ctx, stderr, urlRE)
+
+				if watchURL == "" {
+					t.Error(errors.New(string(stderr.Bytes())))
+				}
+				stderr.Reset()
```

Comment:
hmm it looks like there could be a race between the broadcasting update and this reset+the rest of the test. e.g. adding a time.Sleep(time.Second) before this and logging has

```
    main_test.go:640: success: listening on http://127.0.0.1:50170
        info: compiling index.d2...
        success: successfully compiled index.d2 to index/cream.svg in 52.002ms
        success: successfully compiled index.d2 to index/index.svg in 51.4785ms
        info: broadcasting update to 0 clients
```

I think if you writeFile with an empty file, then connect the client, then writeFile the test d2 after the server is up, it should correctly show the `broadcasting update to 1 client` log avoiding the race on the initial compile.

Reply:
You're saying that the race results in those extra logs right which won't get reset?

It won't matter, we look for the right thing later (broadcasting to 1 client).

Reply:
in this race those logs are present before the reset

Reply:
When we make the page request a couple lines down, it will recompile and rebroadcast there.

Reply:
oh interesting I didn't expect clicking the link to cause a recompile when it hasn't changed, is that a bug?

Reply:
I never questioned it. I guess the alternative is to read the already-written files.

The only instance I can think of where that goes awry is if the user compiles, moves the files, then refreshes their browser or navigates with their browser. 

Worth considering: https://github.com/terrastruct/d2/issues/1708